### ISSUE TITLE
any update?

### ISSUE THREAD

**Issue by brunobahri** [dgunay/galaxy-book4-pro-reverse-engineering#1]
Hi! Thx for this repo 

Any update about this problem?

Im using a Galaxy Book 4 Pro on Fedora 40 and Kernel 6.9.7-200.fc40.x86_64, and both mic and cam didnt work. Its really anoying.

**Comment by PowerKiKi** [comment]
Are you referring to the integrated mic not working ? or a jack mic not working ?

**Comment by brunobahri** [comment]
> Are you referring to the integrated mic not working ? or a jack mic not working ?

Oh, I Mixed it up.

I want to say on board sound**

**Comment by PowerKiKi** [comment]
And what about the webcam ? is it also not working ?

**Comment by brunobahri** [comment]
> And what about the webcam ? is it also not working ?

Yep, doesn't work at all

I've tried Gentoo, Manjaro, Ubuntu, Fedora and Mind, and yesterday, send it and took my money back.

**Comment by PowerKiKi** [comment]
I just learned that the webcam of the Galaxy Book 4 Pro has been fixed via https://github.com/intel/ipu6-drivers/pull/230

**Comment by dgunay** [comment]
No update on the speakers. I returned the laptop weeks ago and frankly I ran out of ideas, being a complete novice at this sort of technical issue.

**Comment by joshuagrisham** [comment]
Hi @brunobahri ,
If you want I can try to help  you get this working. We just finished submitting a patch that should be working for many GB2 and GB3 models and my guess is that it is quite similar for the GB4.
However based on the RtHDDump files it does not seem exactly the same (different coef indexes and possibly even different speaker amp IDs ??).

Is it possible for someone to run a trace using my version of QEMU from here?  
https://github.com/joshuagrisham/galaxy-book2-pro-linux/tree/main/sound/qemu

(more info here: https://github.com/thesofproject/linux/issues/4055#issuecomment-2323411911)

**Comment by dginhoux** [comment]
Hi, 

Thank @joshuagrisham for your help. Its hard to understand how sound work in linux.
I've just ordered a GB4Pro and its the first time i get a problem like that.

I'm trying to understand what are verbs, pins, etc... and finding a way in win11 to get value needed in linux driver.

**Comment by dginhoux** [comment]
There a way ton run hda-analyser with python3, thank to the dev : https://github.com/gm-vm/hda-analyzer

**Comment by BioPho17** [comment]
> Hi @brunobahri , If you want I can try to help you get this working. We just finished submitting a patch that should be working for many GB2 and GB3 models and my guess is that it is quite similar for the GB4. However based on the RtHDDump files it does not seem exactly the same (different coef indexes and possibly even different speaker amp IDs ??).
> 
> Is it possible for someone to run a trace using my version of QEMU from here? https://github.com/joshuagrisham/galaxy-book2-pro-linux/tree/main/sound/qemu
> 
> (more info here: [thesofproject/linux#4055 (comment)](https://github.com/thesofproject/linux/issues/4055#issuecomment-2323411911))

Hi @joshuagrisham, I'm trying to get the speakers working on my Galaxy Book4 Pro 360 (np960qgk), and a lot paths lead to you and your project. I would be happy to run the trace, but I'd need some directions (I made my Github acc today, not super savvy). I'm happy to help in what ways i can though. Thanks for your work!

**Comment by dginhoux** [comment]
Like you @BioPho17, i need some help to understand how to start and find value...

**Comment by stephanlensky** [comment]
@joshuagrisham thanks so much for your work so far on these laptops! Like all of the others on this thread, I am also new to kernel/driver development, and I found your repository for the GB2 very helpful.

Based on your suggestion and some other information I found (like https://jcs.org/2018/11/12/vfio), it seems like our first goal should be getting the speakers to work in a virtualized Windows environment.

I was able to successfully compile a custom Qemu and pass through my PCI device to Windows, but unfortunately it seems like it wasn't enough to get the speakers working, even in the VM. Since audio in the VM isn't working (and actually, the driver doesn't even seem to load properly) I didn't bother going through the full process of dumping the HDA verbs.

I'm not sure what I did wrong, but like the original creator of this repo I put my findings in a repository here in case anyone else finds it helpful: https://github.com/stephanlensky/galaxy-book4-360-linux

I did all of my testing on a Galaxy Book 4 360 (NP960QGK) on Arch Linux.

I still have the laptop and am a bit more comfortable with technical things, so @joshuagrisham feel free to let me know if there is anything obvious I should try. I know it's a little difficult to debug these things over the internet though, so no worries if you'd rather leave it for someone else to pick up 😄

**Comment by BioPho17** [comment]
@Andycodeman  Thank you so much for your work! Hopefully, I can give it a try this weekend.

So, in layman's terms, if I have secure boot disabled, I should just run the quick install command and be good to go?

**Comment by BioPho17** [comment]
> > [@Andycodeman](https://github.com/Andycodeman) Thank you so much for your work! Hopefully, I can give it a try this weekend.
> > So, in layman's terms, if I have secure boot disabled, I should just run the quick install command and be good to go?
> 
> Correct in theory - though I have secure boot enabled and haven't tested with it disabled. So if you decide to try it let me know how it goes. :) Thanks!

Excellent! I'll try a few distros and let you know!

**Comment by dginhoux** [comment]
I've try the work of @Andycodeman on my pro4 14 and all is fully working about the sound. It's the first time since i own this computer i ear sound from theses speakers.

i'm also testing it's good work for the webcam, i've a error with mainline kernel and building v4l2 things

**Comment by BioPho17** [comment]
@Andycodeman I have tried two distros so far.

Galaxy Book pro 360, Secure Boot OFF

Garuda Linux, I had no luck with. I tried various commands with some help with Google Gemini, no luck. I think it uses a "Zen" kernal though, at least that's what is listed when I run uname -r

Good news though, Pop_! OS 24.04 (6.17.9) worked immediately. All I did was run the command and reboot. Not sure if it's because it's a debian/Ubuntu based distro or what, but it was a relief.

Let me know if there's anything else you want me to try. I'm not sure if theres a way to dm, or if you just want to keep this discussion here.

**Comment by Andycodeman** [comment]
@BioPho17 - Thanks for the update!

Yes, Pop OS is Ubuntu based so definitely why it worked fine.  And yes, Garuda is Zen kernel based (I've never even tried this distro to be honest) which is most likely why it failed as it requires the Zen headers in order to successfully build DKMS against the Zen kernel.  Everything in the installer is Linux distro-agnostic except for DKMS.

You could try the following for Garuda...
sudo pacman -S dkms linux-zen-headers i2c-tools

And then re-run the installer to see if that works.

As for dm'ing me, feel free to use Discord @Andycodeman 
But for anything that might help others it would be good to keep here of course. :)
